flytekit/remote/remote.py

```diff
@@ -221,6 +221,34 @@ def file_access(self) -> FileAccessProvider:
         """File access provider to use for offloading non-literal inputs/outputs."""
         return self._file_access
 
+    def get(self, flyte_url: typing.Optional[str] = None) -> typing.Optional[typing.Union[LiteralsResolver, str]]:
+        if flyte_url is None:
+            raise user_exceptions.FlyteUserException("flyte_uri cannot be empty")
+        ctx = self._ctx or FlyteContextManager.current_context()
+        try:
+            data_response = self.client.get_data(flyte_url)
+
+            if data_response.HasField("literal_map"):
+                lm = LiteralMap.from_flyte_idl(data_response.literal_map)
+                return LiteralsResolver(lm.literals)
+            elif data_response.HasField("flyte_deck_download_link"):
+                if len(data_response.flyte_deck_download_link.signed_url) == 0:
+                    raise ValueError(f"Flyte url {flyte_url} resolved to empty download link")
+                d = data_response.flyte_deck_download_link.signed_url[0]
+                remote_logger.debug(f"Attempting to download {d} resolved from flyte url {flyte_url}")
+                fs = ctx.file_access.get_filesystem_for_path(d)
+                with fs.open(d, "rb") as r:
```

Comment:
how about we return a open file object instead of writing to local disk?

Reply:
I would love if we could put this block in here
```python
    raw_html = fs.read()
    if not ignore_jupyter and _ipython_check():
        from IPython.core.display import HTML

        return HTML(raw_html)
    return raw_html
```
https://ipython.readthedocs.io/en/stable/api/generated/IPython.display.html

I do not like writing to files. Let the users write it to file if they want

This is implemented already in flytekit and is very useful
https://github.com/flyteorg/flytekit/blob/8e5bf6caaf7f2facee8eb24a67d6a5d93d9c7019/flytekit/deck/deck.py#L93
returned here
https://github.com/flyteorg/flytekit/blob/8c05797feeaf0e47a56075bf1c725c8284157660/flytekit/core/context_manager.py#L639


maybe move this to utils
https://github.com/flyteorg/flytekit/blob/8e5bf6caaf7f2facee8eb24a67d6a5d93d9c7019/flytekit/deck/deck.py#L77

Reply:
done